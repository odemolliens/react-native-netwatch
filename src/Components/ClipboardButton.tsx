import * as React from 'react';
import { useContext } from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import { ThemeContext } from '../Theme';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

export interface IProps {
  onPress: () => void;
}

// Button use in the details screens to copy to clipboard a text. It's just a button text.
export const ClipboardButton: React.FC<IProps> = (props: IProps) => {
  const theme = useContext(ThemeContext);
  return (
    <TouchableOpacity testID="buttonCopyToClipboard" style={styles.container} onPress={props.onPress}>
      <MaterialCommunityIcons name="clipboard-arrow-left-outline" color={theme.primaryColor} size={14} />
      <Text style={[{ color: theme.primaryColor, marginLeft: 6 }]}>Copy</Text>
    </TouchableOpacity>
  );
};

export default ClipboardButton;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    borderLeftWidth: 0,
    alignItems: 'center',
  },
});
