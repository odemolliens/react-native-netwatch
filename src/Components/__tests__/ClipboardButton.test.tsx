import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { ClipboardButton, IProps } from '../ClipboardButton';

describe('Status test suite', () => {
  let component: ShallowWrapper;
  let props: IProps;
  const onPress = jest.fn();

  it('should render properly redux action', () => {
    givenProps(onPress);
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  // GIVEN
  function givenComponent() {
    component = shallow(<ClipboardButton {...props} />);
  }

  function givenProps(onPress: () => void) {
    props = {
      onPress,
    };
  }
});
