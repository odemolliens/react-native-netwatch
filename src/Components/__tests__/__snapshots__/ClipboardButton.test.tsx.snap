// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status test suite should render properly redux action 1`] = `
<ForwardRef
  onPress={[MockFunction]}
  style={
    Object {
      "alignItems": "center",
      "borderLeftWidth": 0,
      "flexDirection": "row",
    }
  }
  testID="buttonCopyToClipboard"
>
  <Icon
    allowFontScaling={false}
    color="#06B6D4"
    name="clipboard-arrow-left-outline"
    size={14}
  />
  <Component
    style={
      Array [
        Object {
          "color": "#06B6D4",
          "marginLeft": 6,
        },
      ]
    }
  >
    Copy
  </Component>
</ForwardRef>
`;
